infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/runtime/LogReplicationServerRouter.java

```diff
@@ -26,11 +23,11 @@
 import java.util.Map;
 import java.util.Optional;
 
+
 /**
- * This class represents the Corfu interface to route incoming messages from external adapters when
- * custom communication channels are used.
- *
- * Created by annym on 14/5/20.
+ * This class is shared by all the connection receiver routers.
```

Comment:
I am somehow confused on what this class is doing. It used to have a handler map which 'routed' the request to the correct server (which in our case was really always the LogReplicationServer). Then what I imagined was that the server would treat accordingly per session.

Now, from what I'm understanding your approach was a bit different, there is a Router per session, so what is this router doing?

Reply:
I meant to remove some methods in this class during dev, but seemed to have missed it. Thanks.

I have removed this file now.
The IServerRouter in this class was not needed, and after removing that, the left over code could be moved to the interClusterServerNode.

The code moved to interClusterServerNode is essentially creating and starting the transport server. Since there can only be 1 server per service, we need all the routers to interact with the same transport server.